include/deal.II/particles/property_pool.h

```diff
@@ -0,0 +1,88 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__particles_property_pool_h
+#define dealii__particles_property_pool_h
+
+#include <deal.II/base/array_view.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Particles
+{
+  /**
+   * This class manages the memory space in which particles store their
+   * properties. Because this is dynamic memory and every particle needs the
+   * same amount it is more efficient to let this be handled by a central
```

Comment:
out of curiosity: is it limited to the same amount per particle? What if particles have different types and thus needs to store their properties?